### ISSUE TITLE
Exception thrown by upload server

### ISSUE THREAD

**Issue by csegarragonz** [faasm/faasm#634]
The distributed tests are [failing](https://github.com/faasm/faasm/runs/5873019369?check_suite_focus=true) in the CI with an error during function upload. I can reproduced the exception locally.

The minimal set up to reproduce the exception from the current main:

```bash
source ./bin/workon.sh
inv cluster.start

# Build the latest upload
./bin/cli.sh faasm
inv dev.cc upload
exit

# Check upload logs
docker-compose logs -f upload

# From a different terminal
./bin/cli.sh cpp

inv func demo hello

# This will eventually fail
inv func.upload-user demo
```

**Comment by csegarragonz** [comment]
The error seems to have escaped GHA because in the distributed or quick start tests we don't use the cluster tasks to create the cluster, rather call docker-compose directly. This means that we use whatever version of the code we shipped with the latest container image, effectively skipping commits until a new release comes around.

**Comment by Shillaker** [comment]
The issue with using the current release in the quick-start tests is already covered by this issue: https://github.com/faasm/faasm/issues/533 

The issue related to the Faabric server should be fixed here: https://github.com/faasm/faabric/pull/254

**Comment by csegarragonz** [comment]
The issue I was seeing was fixed in faasm/faabric#254 so will close now.
